### PR TITLE
Add timing of profiles to StructuredProfiler and UnstructuredProfiler

### PR DESCRIPTION
**Pull request by JGSweets**
Adds timeit to `StructuredProfiler` and `UnstructuredProfiler` for the following calculations in each.
StructuredProfiler: correlations, row stats
UnstructuredProfiler: getting null values and base stats

### CODE REVIEW

File: dataprofiler/tests/profilers/utils.py [hunk 1]
```diff
@@ -132,3 +133,17 @@ def clean_report(report):
...
+    Creates a mock for the time.time function that increments the time for
+    every call.
+    """
+    def increment_counter():
+        counter = 0
+        while True:
+            counter += 1
+            yield counter
+    counter = increment_counter()
+    return mock.patch('time.time', side_effect=lambda: next(counter))
```
Comment (gme5078) [inline dataprofiler/tests/profilers/utils.py:149]: nifty.

File: dataprofiler/profilers/profile_builder.py [hunk 2]
```diff
@@ -970,15 +974,15 @@ def report(self, report_options=None):
...
             }),
             ("data_stats", OrderedDict()),
         ])
-
         report["data_stats"] = self._profile.profile
         return _prepare_report(report, output_format, omit_keys)
 
-    @staticmethod
-    def _clean_data_and_get_base_stats(data, sample_size,
+    @utils.method_timeit(name="clean_and_base_stats")
```
Comment (gme5078) [inline dataprofiler/profilers/profile_builder.py:984]: I'm approving, but I dont think the average user will understand clean_and_base_stats

**Review by gme5078 — APPROVED** [review]

File: dataprofiler/profilers/profile_builder.py (hunk 2 above)
Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py:984]: i'm open to other names, i wasn't really sure what to put.

Comment (AnhTruong) [reply dataprofiler/profilers/profile_builder.py:984]: may be `get_clean_and_base_stats `

File: dataprofiler/profilers/profile_builder.py [hunk 3]
```diff
@@ -1490,6 +1498,7 @@ def _update_correlation(self, clean_samples, prev_dependent_properties):
             batch_corr, batch_properties["mean"],
             batch_properties["std"], batch_samples)
 
+    @utils.method_timeit(name='correlation')
```
Comment (AnhTruong) [inline dataprofiler/profilers/profile_builder.py:1501]: should we make the name differently as we had correlation above already?
Also, how we name other stats timing to distinguish between update and merge?

Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py:1501]: we don't separate the cost between merge / update of correlations. All together, they are the cost of computing the statistic.

File: dataprofiler/profilers/profile_builder.py (hunk 2 above)
Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py:984]: I'm not sure that makes it more clear to the user as the intent is to label the cost of what is being calculated but not just the function

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by gme5078 — APPROVED** [review]

**Review by ChrisWallace2020 — APPROVED** [review]

**Pull request merged**